From 4bbf42f1841873f38014ad8500c68fb95e80960d Mon Sep 17 00:00:00 2001
From: Arturo Rinaldi <arty.net2@gmail.com>
Date: Mon, 6 Feb 2017 12:51:15 +0100
Subject: [PATCH 4/4] fixing reset stm32 address print to screen

---
 src/main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/main.c b/src/main.c
index 2d5aa92..632b745 100644
--- a/src/main.c
+++ b/src/main.c
@@ -707,7 +707,7 @@ status_again:
 		//ST Application Note 3156 Documents how to reset an STM32 out of DFU mode and into firmware mode
 		//Basicly, send the target vector reset address, then a zero-length download command, then by a get status command.

-		printf("Resetting STM32, starting firmware at address 0x%X...\n",vector_address);
+		printf("Resetting STM32, starting firmware at address 0x0%X...\n",vector_address);
 		int otto_ret = dfuse_special_command(dfu_root, vector_address, SET_ADDRESS);
 		if( otto_ret < 0 ) {
 			printf("Error: Unable to set start address for reseting\n");
--
2.11.0
