--- a/deps/uv/src/unix/getaddrinfo.c	2015-10-29 13:22:04.000000000 +0100
+++ b/deps/uv/src/unix/getaddrinfo.c	2015-11-14 18:16:22.184081502 +0100
@@ -99,6 +99,7 @@
   int err;
 
   req = container_of(w, uv_getaddrinfo_t, work_req);
+	req->hints->ai_flags &= ~AI_V4MAPPED;
   err = getaddrinfo(req->hostname, req->service, req->hints, &req->addrinfo);
   req->retcode = uv__getaddrinfo_translate_error(err);
 }
