--- a/lib/module.js	2015-10-29 13:22:06.000000000 +0100
+++ b/lib/module.js	2015-11-14 18:21:00.285261212 +0100
@@ -587,7 +587,8 @@
     var homeDir = process.env.HOME;
   }

-  var paths = [path.resolve(process.execPath, '..', '..', 'lib', 'node')];
+  var paths = [path.resolve(process.execPath, '..', '..', 'lib', 'node'),
+	 path.resolve(process.execPath, '..', '..', 'lib', 'node_modules')];

   if (homeDir) {
     paths.unshift(path.resolve(homeDir, '.node_libraries'));
