Index: octave-3.2.4/configure
===================================================================
--- a/configure	2011-01-02 15:14:21.000000000 +0100
+++ b/configure	2011-01-02 15:14:32.000000000 +0100
@@ -17606,7 +17606,7 @@
 $as_echo "libraries $x_libraries, headers $x_includes" >&6; }
 fi

-if test "$have_x" = yes; then
+if test "$have_x" = "yes"; then

 $as_echo "#define HAVE_X_WINDOWS 1" >>confdefs.h

Index: octave-3.2.4/configure.ac
===================================================================
--- a/configure.ac	2016-03-21 20:32:59.000000000 +0100
+++ b/configure.ac	2016-04-07 19:23:14.798327863 +0200
@@ -1139,7 +1139,7 @@
 ### Check for X11 libraries

 AC_PATH_X
-if test "$have_x" = yes; then
+if test "$have_x" = "yes"; then
   AC_DEFINE(HAVE_X_WINDOWS, 1, [Define to 1 if you have X11.])

   if test "$x_includes" != "NONE"; then
